enhancements/external-control-plane-topology.md

```diff
@@ -0,0 +1,93 @@
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+External control plane support was introduced in OCP 4 with the support
+for the [IBM Cloud Managed service](https://github.com/openshift/enhancements/blob/master/enhancements/update/ibm-public-cloud-support.md) (ROKS). At the time, this was the only platform that
+was run with an external control plane. It was sufficient to use a platform type of
+`IBMCloud` to distinguish it from other OCP installations with traditional
+control plane topology.
+
+More recently, an orthogonal field was [added to the Infrastructure resource](https://github.com/openshift/enhancements/blob/master/enhancements/single-node/cluster-high-availability-mode-api.md) to indicate
+the type of control plane topology. Current supported values for the `controlPlaneTopology` field
+are `HighlyAvailable` and `SingleReplica`.
+
+This enhancement proposes adding a third possible value to the control plane topology
+field: `External`. A value of `External` in this field indicates that control plane components
+such as Etcd, Kube API server, Kube Controller Manager, and Kube Scheduler are running outside
+the cluster and are not visible as pods inside the cluster.
+
+## Motivation
+
+Whether the control plane is external or not should not be tied to the platform that the cluster
+is running on. IBM Cloud will soon support IPI/UPI installation. Thus, having a platform of `IBMCloud`
+will not imply an external control plane. Hypershift will bring support for external control planes
+to existing platforms such as AWS.
+
+### Goals
+
+- Allow expressing a new type of control plane topology in the Infrastructure resource inside an
+  OCP cluster.
+
+- Provide operators/components that change their behavior based on whether the control plane is external or
+  self hosted, a clear indicator of what mode they're running in.
+
+### Non-Goals
+
+- Provide a design for running OCP with an external control plane.
+
+- Describe how the `controlPlaneTopology` field will be set for an external control plane deployment.
+
+## Proposal
+
+- Add `External` as an additional option to the `TopologyMode` type for `Infrastructure`
+
+### Implementation Details
+
+Currently `External` only makes sense as a mode for the ControlPlaneTopology, not the InfrastructureTopology, but
+both use the same type (`TopologyMode`).
+
+The Hypershift or IBM ROKS installer is responsible for setting this value when bootstrapping a new hosted control
+plane OpenShift cluster.
+
+#### Component Impact
+
+Existing components such as Console and Monitoring that use platform type of `IBMCloud` to modify their behavior
+for external control planes would need to switch to `controlPlaneTopology` to determine whether the control plane
+is external.
```

Comment:
Is the test plan for this "CI jobs with an external control plane pass e2e without blowing up"?  Do we have CI coverage for things like whatever console twiddles based on this (looks like maybe just [dashboard availability][1])?  Or is the expectation that we make a best-effort attempt to transition existing `IBMCloud` consumers and then fix any we miss (or which slip in later) as we stumble across them?

[1]: https://github.com/openshift/console/blob/b7ff8039b6971e5ed30a5d3f4ecfbefce6d617a9/frontend/packages/console-app/src/plugin.tsx#L125-L140

Reply:
Yes, I didn't think it was relevant to explain that Hypershift or IBM ROKS toolkit do their own bootstrapping and will set the value appropriately. But if you think it's relevant I can add it to the doc.

Reply:
The test plan for IBM Cloud is that QE on their side will verify that we have not regressed in the tweaks that have been made based on the `IBMCloud` platform. On our side, we don't yet have automated CI for Hypershift setups, but it's something that we're working towards.